cmd/bankapp/main.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"fmt"
+	"github.com/gorilla/mux"
+	"github.com/iliyanmotovski/bankv2/bank/api"
+	"github.com/iliyanmotovski/bankv2/bank/domain"
+	"github.com/iliyanmotovski/bankv2/bank/persistence"
+	"github.com/justinas/alice"
+	"gopkg.in/mgo.v2"
+	"net/http"
+	"time"
+)
+
+func main() {
+	session, err := mgo.DialWithTimeout("localhost:27016", time.Second*10)
+	if err != nil {
+		fmt.Println(err)
+	}
+	defer session.Close()
+	session.SetMode(mgo.Monotonic, true)
+
+	renew := domain.Renew{RenewSessionCounter: make(chan bool)}
+
+	r := mux.NewRouter()
+
+	mongoSessionRepo := persistence.NewSessionRepository(*session, "bank")
+
+	mongoImplementedUserStore := domain.UserStore(mongoSessionRepo)
+	mongoImplementedUserSessionStore := domain.SessionStore(mongoSessionRepo)
+	mongoImplementedUserAccountStore := domain.AccountStore(mongoSessionRepo)
+
+	SignUpHandlers := alice.New(api.LoggingMiddleware)
+	LoginLogoutHandlers := alice.New(api.LoggingMiddleware, api.RecoverMiddleware, api.CheckValidSessionOnLoginAndLogoutMiddleware(mongoImplementedUserSessionStore))
+	TransactionHandlers := alice.New(api.LoggingMiddleware, api.RecoverMiddleware, api.CheckValidSessionOnTransactionMiddleware(mongoImplementedUserSessionStore, mongoImplementedUserAccountStore, renew))
+	TrHistoryHandlers := alice.New(api.LoggingMiddleware, api.RecoverMiddleware, api.CheckValidSessionOnHistoryRequestMiddleware(mongoImplementedUserSessionStore, renew))
+
+	r.Handle("/", http.FileServer(http.Dir("./bank/static")))
+	r.Handle("/users/signup", SignUpHandlers.Then(api.SignUpHandler(mongoImplementedUserStore)))
+	r.Handle("/users/login", LoginLogoutHandlers.Then(api.LoginHandler(mongoImplementedUserSessionStore, mongoImplementedUserStore, mongoImplementedUserAccountStore, renew)))
```

Comment:
You can use subrouter to eliminate the duplication of users and attaching of the same security middleware:

```golang
	usersRouter := router.NewRoute().Subrouter()

```